pkg/controller/cloud/nodecontroller.go

```diff
@@ -91,10 +109,79 @@ func (cnc *CloudNodeController) Run() {
 	go func() {
```

Comment:
all this goroutine does is spawn 2 more goroutines - can we cull the first level of goroutine?